### PR TITLE
EVMJIT: better jumps resolving

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: evmjit/libevmjit/Compiler.cpp [hunk 1]
```diff
@@ -93,21 +95,40 @@ std::vector<BasicBlock> Compiler::createBasicBlocks(code_iterator _codeBegin, co
...
+		auto term = it->getTerminator();
 
-	// TODO: Extend this function as `resolveJumps()` and fill gaps in branch instructions.
-	auto target = llvm::cast<llvm::PHINode>(m_jumpTableBB->begin());
-	for (auto pred: llvm::predecessors(m_jumpTableBB))
-	{
-		auto targetMd = llvm::cast<llvm::LocalAsMetadata>(pred->getTerminator()->getMetadata("target")->getOperand(0));
-		target->addIncoming(targetMd->getValue(), pred);
+		if (!term)
+		{
```
Comment (gavofyork) [inline evmjit/libevmjit/Compiler.cpp]: unneeded `{`s

File: evmjit/libevmjit/Compiler.cpp [hunk 2]
```diff
@@ -93,21 +95,40 @@ std::vector<BasicBlock> Compiler::createBasicBlocks(code_iterator _codeBegin, co
...
+			// Block may have no terminator if the next instruction is a jump destination.
+			llvm::IRBuilder<>{it}.CreateBr(nextBlock);
+		}
+		else if (auto jump = llvm::dyn_cast<llvm::BranchInst>(term))
+		{
```
Comment (gavofyork) [inline evmjit/libevmjit/Compiler.cpp]: unneeded `{`s.

**Comment by gavofyork** [comment]
minor style issues.

**Comment by chfast** [comment]
Oh no this one again :D 
Is it really mandatory? I find it looking much better when using {} for body longer than 1 line.

**Pull request merged**